src/core/toga/sources/list_source.py

```diff
@@ -98,8 +98,9 @@ def append(self, *values, **named):
         return self.insert(len(self), *values, **named)
 
     def remove(self, row):
+        index = self._data.index(row)
         self._data.remove(row)
-        self._notify('remove', item=row)
+        self._notify('remove', index=index, item=row)
```

Comment:
I'm not sure this is the best change (and I will also have to fix a few texts if we keep this). It seems that the way to remove an item in gtk is to [provide the index rather than the value](https://lazka.github.io/pgi-docs/#Gtk-3.0/classes/ComboBoxText.html#Gtk.ComboBoxText.remove).

The reason this is necessary is because the ListSource calls the `_notify` method after a removal, so there's no way to know the index of the removed item after the fact.